### PR TITLE
Module2

### PR DESCRIPTION
**Pull request by MariyaKuksar**

### CODE REVIEW

File: gift-certificates-system/persistence/src/main/java/com/epam/esm/configuration/PersistenceConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
+package com.epam.esm.configuration;
+
+import com.zaxxer.hikari.HikariDataSource;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.*;
```
Comment (ShiniTada) [inline gift-certificates-system/persistence/src/main/java/com/epam/esm/configuration/PersistenceConfiguration.java]: please avoid using imports with .*

File: gift-certificates-system/service/src/main/java/com/epam/esm/service/impl/TagServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        List<Tag> tagList = tagDao.findAll();
+        return tagList.stream()
+                .map(tag -> modelMapper.map(tag, TagDto.class))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public void deleteTag(long id) {
+        tagValidator.validateId(id);
+        if (!tagDao.delete(id)) {
```
Comment (ShiniTada) [inline gift-certificates-system/service/src/main/java/com/epam/esm/service/impl/TagServiceImpl.java:75]: Will the record from the joining table with certificates be removed after that action?
If yes, that what will happen with certificates, which already have that tag attached?

**Review by IvanYarochkin — APPROVED** [review]

File: gift-certificates-system/persistence/src/main/java/com/epam/esm/configuration/PersistenceConfiguration.java (hunk 1 above)
Comment (MariyaKuksar) [reply gift-certificates-system/persistence/src/main/java/com/epam/esm/configuration/PersistenceConfiguration.java]: ok, I fixed and commited it

File: gift-certificates-system/service/src/main/java/com/epam/esm/service/impl/TagServiceImpl.java (hunk 2 above)
Comment (MariyaKuksar) [reply gift-certificates-system/service/src/main/java/com/epam/esm/service/impl/TagServiceImpl.java:75]: Yes, I use  MySQL  ON DELETE CASCADE for gift_certificate_tag_connection table, the record from the joining table with certificates be removed, the certificate will exist without this tag.

**Review by ShiniTada — APPROVED** [review]

**Pull request merged**